packages/data-context/src/actions/ProjectActions.ts

```diff
@@ -1,5 +1,6 @@
 import type { MutationAppCreateConfigFileArgs } from '@packages/graphql/src/gen/nxs.gen'
 import type { FoundBrowser, FullConfig, LaunchArgs, LaunchOpts, OpenProjectLaunchOptions } from '@packages/types'
+import { getProjectRoots, insertProject } from '@packages/server/lib/cache'
```

Comment:
This is probably okay fo now but we should sync with @tgriesser - ideally the "new" code should not be pulling in existing code. I think the ideal workflow will be any time we need to access untyped code, we should pull it into a new package and strongly type it.

In practice, it's impractical if we want to ship anything in the near future, since the majority of `server` is untyped, but we should definitely have a shared understanding of the eventual goal w.r.t type safety moving forward. I'm having a similar problem in another feature I'm working on, so it'd be good to sync on this.